#1666 Fixed bug setDefaultValues option clears all the existing properties

nmgokhale opened the pull request:
Fixes: #1666 

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

matthoward366 reviewed (approved):
I'll approve this but can we have a followup PR with unit tests?

nmgokhale commented:
I added a unit test for this when I added this feature - https://github.com/elyra-ai/canvas/blob/main/canvas_modules/common-canvas/__tests__/common-properties/properties-controller-test.js#L1197

nmgokhale commented:
Oh I will update the test to check for existing properties after setting` setDefaultValues: true`. Adding another commit to this PR.
